x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_creation/engine_creation_logic.test.ts

```diff
@@ -63,6 +64,28 @@ describe('EngineCreationLogic', () => {
         expect(EngineCreationLogic.values.name).toEqual('name-with-special-characters');
       });
     });
+
+    describe('submitEngine', () => {
+      it('sets isLoading to true', () => {
```

Comment:
I would have put this with the `submitEngine` listener below, but doing so caused other tests to fail because of the way this test suite doesn't automatically clear mocks between each test 🤷

Reply:
I think `submitEngine` should be clearing mocks before each test. There's certain times where using a `beforeAll` is appropriate as a performance optimization, but I don't think this is one of those.

Reply:
Same TBH, but I didn't want to muddy this PR too much or make a large set of unrelated changes 🤔 I'll think about a separate refactor at some point for the files if that's cool